src/main/java/org/jenkinsci/test/acceptance/docker/DockerContainerHolder.java

```diff
@@ -22,15 +25,35 @@
     @Inject
     Docker docker;
 
+    @Inject
+    private FailureDiagnostics diag;
+
     T container;
 
     /**
+     * Injecting a portOffset will force the binding of dockerPorts to local Ports with an offset
+     * (e.g. bind docker 22 to localhost port 40022,
+     */
+    @Inject(optional = true)
+    @Named("dockerPortOffset")
+    private int portOffset = 0;
+
+    /**
      * Lazily starts a container and returns the instance.
      */
     @Override
     public synchronized T get() {
-        if (container==null)
-            container = docker.start((Class<T>)type.getRawType());
+        if (container==null) {
+            Class<T> fixture = (Class<T>) type.getRawType();
+            File buildlog = diag.touch("docker-" + fixture.getSimpleName() + ".build.log");
```

Comment:
so this change means a failure to build an image for a test will not be in the junit log for that specific test, so you have to go looking in multiple places.
Therefore to get this log we now need to archive all these logs, and this means for the case of successful building we will not throw these logs away (like we can do with the junit trim successful tests output).